### PR TITLE
Missing functions in Ship class.

### PR DESCRIPTION
**Pull request by ArturGubala**
Added function:
- getCargo,
- getAllCargos.

Refactored function countAvailableSpace in class Player.

### CODE REVIEW

**Review by ArturGubala — COMMENTED** [review]
My auto format broke previous format in Ship.hpp, I'll fix it soon.

File: shm/Player.hpp [hunk 1]
```diff
@@ -13,6 +13,7 @@ class Player {
     size_t getAvaliableSpace() const { return avaliableSpace_; };
     size_t getSpeed() const { return ship_->getSpeed(); };
     Cargo* getCargo(size_t index) const { return ship_->getCargo(index); };
+    std::vector<std::shared_ptr<Cargo>> getAllCargos() const { return ship_->getAllCargos(); }
```
Comment (J-Cieplinski) [inline shm/Player.hpp]:
```suggestion
    std::vector<std::shared_ptr<Cargo>>& getAllCargos() const { return ship_->getAllCargos(); }
```

So it's not copied every time

**Review by J-Cieplinski — CHANGES_REQUESTED** [review]

**Comment by J-Cieplinski** [comment]
> My auto format broke previous format in Ship.hpp, I'll fix it soon.

Don't bother. This one is good

File: shm/Player.hpp (hunk 1 above)
Comment (ArturGubala) [reply shm/Player.hpp]: Yes, yes - thanks!

Comment (ArturGubala) [reply shm/Player.hpp]: But after add reference i got this error "initial value of reference to non-const must be an lvalue" and I don't have idea why.

Comment (J-Cieplinski) [reply shm/Player.hpp]: Possibly method which uses this function is not declared as const

Comment (ArturGubala) [reply shm/Player.hpp]: When I write it like this: `const std::vector<std::shared_ptr<Cargo>>&` both in Ship and Player there is no error, but it's correct?

Comment (J-Cieplinski) [reply shm/Player.hpp]: I would say it's ok since it's just a getter

File: shm/Ship.hpp [hunk 2]
```diff
@@ -9,27 +9,21 @@
...
 
-    size_t getCapacity() const  { return capacity_; }
-    size_t getMaxCrew() const   { return maxCrew_; }
-    size_t getSpeed() const     { return speed_; }
+    size_t getCapacity() const { return capacity_; }
+    size_t getMaxCrew() const { return maxCrew_; }
+    size_t getSpeed() const { return speed_; }
     std::string getName() const { return name_; }
-    size_t getId() const        { return id_; }
+    size_t getId() const { return id_; }
```
Comment (niemarcin) [inline shm/Ship.hpp]: You've done changes related to code style in Ship.cpp file but you've added it to a commit "add functions which return particular cargo and all cargos, refactor Player function countAvailableSpace". It would be better do split these changes into two commits so it's easier to go through the project history and revert some changes if needed later.

Comment (niemarcin) [inline shm/Ship.hpp]: Btw. Why did you change those lines? Is it related to comment in PR with our OOP_homework1 ? About keeping one format?

File: shm/Player.cpp [hunk 3]
```diff
@@ -9,8 +9,8 @@ Player::Player(std::unique_ptr<Ship> ship, size_t money) {
 };
 
 size_t Player::countAvailableSpace() {
-    auto occupiedSpace = std::accumulate(begin(ship_->getAllCargos()),
-                                         end(ship_->getAllCargos()),
+    auto occupiedSpace = std::accumulate(begin(getAllCargos()),
+                                         end(getAllCargos()),
```
Comment (niemarcin) [inline shm/Player.cpp]: Actually I prefer the previous version - for me it's easier to understand that I want to get AllCargos from ship when I see `ship_->getAllCargos()`. It's just explicit way to say "get cargo from ship object". And `getAllCargos()` it's like - "okay, get cargo but from where?".

File: shm/Player.cpp [hunk 4]
```diff
@@ -9,8 +9,8 @@ Player::Player(std::unique_ptr<Ship> ship, size_t money) {
...
                                          0,
                                          [](const Cargo& cargoFirst, const Cargo& cargoSecond) {
                                              return cargoFirst.getAmount() + cargoSecond.getAmount();
```
Comment (niemarcin) [inline shm/Player.cpp]: Are you sure this is a proper usage of std::accumulate algorithm? Based on https://en.cppreference.com/w/cpp/algorithm/accumulate the binary operation (lambda in our case) is defined as:

> op - binary operation function object that will be applied. The binary operator takes the current accumulation value a (initialized to init) and the value of the current element b. The signature of the function should be equivalent to the following: `Ret fun(const Type1 &a, const Type2 &b);` 

So it seems that the first parameter of lambda should be current accumulation value. And current value from Cargo vector is the second parameter of lambda.

**Review by niemarcin — CHANGES_REQUESTED** [review]

File: shm/Player.hpp (hunk 1 above)
Comment (niemarcin) [reply shm/Player.hpp]: I guess that `ship_->getAllCargos(); ` should have the same return type as`getAllCargos() ` in Player class  so both should be `std::vector<std::shared_ptr<Cargo>>&`

File: shm/Ship.hpp (hunk 2 above)
Comment (ArturGubala) [reply shm/Ship.hpp]: It wasn't me, it was my auto format options. I wrote about it just after I pushed changes and noticed that. These format changes are not a part of our PR with our OPP_homewark1. We have other file with different formating style. Did we talk about formating style btw?

File: shm/Player.cpp (hunk 4 above)
Comment (ArturGubala) [reply shm/Player.cpp]: Yeah, my mistake.

**Comment by ArturGubala** [comment]
I did revert. In next steps I'll do commit per function and after we settle format rules I'll format code in that way.

**Review by J-Cieplinski — APPROVED** [review]

**Review by lisie-sprawy — APPROVED** [review]

**Review by iprice73 — APPROVED** [review]

**Review by niemarcin — APPROVED** [review]
